test/cppsuite/test_harness/workload/thread_context.cxx

```diff
@@ -219,21 +219,29 @@ thread_context::update(scoped_cursor &cursor, uint64_t collection_id, const std:
 }
 
 bool
-thread_context::insert(scoped_cursor &cursor, uint64_t collection_id, uint64_t key_id)
+thread_context::insert(
+  scoped_cursor &cursor, uint64_t collection_id, uint64_t key_id, wt_timestamp_t ts)
+{
+    return insert(cursor, collection_id, key_to_string(key_id), ts);
+}
+
+bool
+thread_context::insert(
+  scoped_cursor &cursor, uint64_t collection_id, const std::string &key, wt_timestamp_t ts)
 {
     WT_DECL_RET;
-    std::string key, value;
+    std::string value;
     testutil_assert(tracking != nullptr);
     testutil_assert(cursor.get() != nullptr);
 
     /*
-     * Get a timestamp to apply to the update. We still do this even if timestamps aren't enabled as
-     * it will return a value for the tracking table.
+     * When no timestamp is specified, get one to apply to the update. We still do this even if the
+     * timestamp manager is not enabled as it will return a value for the tracking table.
      */
-    wt_timestamp_t ts = tsm->get_next_ts();
+    if (ts == 0)
```

Comment:
How would you implement a test that doesn't use timestamps?

Reply:
Our current tests do not specify any timestamp, the manager takes care of them. That's why I put a default value of `0`.
Soon, we will implement tests where we need to set arbitrary ts values, we don't want to ts manager to do it for us. See WT-7924 and the related tickets to know more, happy to discuss more.

Reply:
As discussed, I misunderstood your question. We do force the use of timestamps in the cpp framework.

Reply:
All good, thanks! I guess, when the need arises we'll implement the support of no timestamp and mixed-mode.